packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1571,6 +1574,231 @@ abstract class InheritedWidget extends ProxyWidget {
+///   // ...
+///   static MyModel of(BuildContext context, String dependency) {
+///     return context.inheritFromWidgetOfExactType(IntegerModel, token: dependency);
+///   }
+/// }
+/// ```
+///
+/// Calling `MyModel.of(context, 'foo')` means that `context` depends on the
+/// `foo` aspect of `MyModel`.
+///
+/// The [updateShouldNotifyDependent] method decides if a change in the model
+/// warrants rebuilding a dependent (BuildContext). It must compare the old
+/// and new [InheritedModel] widget configurations with a set of dependencies
+/// and decide if the model's changes correspond to any of the dependencies.
+///
+/// For example:
+/// ```
+/// class ABModel extends InheritedModel<String> {
+///   ABModel({ this.a, this.b, Widget child }) : super(child: child);
+///
+///   final int a;
+///   final int b;
+///
+///   @override
+///   bool updateShouldNotifyDependent(ABCModel old, Set<String> dependencies) {
+///     return (a != old.a && dependencies.contains('a'))
+///         || (b != old.b && dependencies.contains('b'))
+///   }
+///
+///   // ...
+/// }
+/// ```
+///
+/// In the previous example the dependency checked by
+/// [updateShouldNotifyDependent] is just the token string passed to
+/// `inheritFromWidgetOfExactType`. Sometimes it's useful to have the
+/// model create a dependency object that's based on the token but takes
+/// the current state of the model into account. The [getDependencyFor]
+/// method is called when a dependency is first created with
+/// `inheritFromWidgetOfExactType` and each time [update] process is
+/// completed.
+///
+/// In the following code example, `IntegerTree` represents a data model
+/// whose values are specified with paths, like '/a/b/c'. The IntegerModel
+/// has-a integer tree instance and is rebuilt when its descendants' view
+/// of the model should be updated. The [getDependencyFor] override
+/// converts the token, which is a path, to a `PathDependency` object. The
+/// path dependency saves the value of the path so that
+/// `updateShouldNotifyDependent` can compare it with the current value.
+///
+/// ```
+/// class IntegerTree {
+///   int valueOf(String path) => _lookup(path);
+///   // ...
+/// }
+///
+/// class PathDependency {
+///   PathDependency(this.path, this.value);
+///   final String path;
+///   final int value;
+/// }
+///
+/// class IntegerModel extends InheritedModel<PathDependency> {
+///   IntegerModel({ this.integerTree, Widget child }) : super(child: child);
+///
+///   final IntegerTree integerTree;
+///
+///   @override
+///   PathDependency getDependencyFor(dynamic token) {
+///     final String path = token;
+///     return new PathDependency(path, integerTree.valueOf(path));
+///   }
+///
+///   @override
+///   bool updateShouldNotifyDependent(IntegerModel old, Set<PathDependency> dependencies ) {
+///     for (PathDependency dependency in dependencies) {
+///       if (dependency.value != integerTree.valueOf(dependency.path))
```

Comment:
why would you want to do this instead of just looking up the path in the old model and comparing it to the value in the new model?

Reply:
Because the widget's model hasn't changed, only the model's internal state has changed.

Reply:
I don't think we should be encouraging people to pass in mutable objects to InheritedWidgets. It's very easy to end up with hard-to-find bugs when doing that, because there's nothing that forces the inherited widget to get updated, making the whole update mechanism very brittle.

Reply:
Can you elaborate? I thought InheritedWidget's reason for existing was to allow UI to depend on mutable state.